### PR TITLE
MVP of hide karma option

### PR DESCRIPTION
**Pull request by jpaddison3**
Implement an option to hide karma from visibility in the UI. It is intentionally fairly tucked away. You have to first enable a user setting, and then separately enable a post setting. And you can only do this if you are `trustLevel1`.

It's noticeably pretty easy to workaround. We do nothing to prevent you from hitting the API to find the karma. Because of this, greaterwrong will not respect this feature. My guess is that this is a good enough nudge. It certainly seems so for the EA Forum (see below), where I don't know of anyone (except maybe Wai Dai, who sometimes visits) who uses it.

I feel ok with users having to disable the karma notifications if they really don't want to see their own karma. It still prevents common knowledge as is.

If you would like, I can turn `hidden` on on LW and AF, and go through the initial testing on the EA Forum.

### CODE REVIEW

**Comment by Discordius** [comment]
Current sense is that it seems probably fine to merge, but would want you to deactivate it on LW for now, and then we can think about whether we want to activate it on LW later, after some discussion with the community

**Comment by jpaddison3** [comment]
@Discordius back to you

**Comment by jpaddison3** [comment]
Gonna go ahead and go with the graphql level change.

File: packages/lesswrong/lib/collections/posts/helpers.js [hunk 1]
```diff
@@ -195,3 +195,13 @@ Posts.getKarma = (post) => {
...
 }
+
+// User can add/edit the hideCommentKarma setting if:
+//  1) The user has the requiset setting enabled
+//  And
+//  2) The post does not exist yet
+//  Or if the post does exist
+//  3) The post doesn't have any comments yet
+Posts.canEditHideCommentKarma = (user, post) => {
+  return user['showHideKarmaOption'] && (!post || !Posts.getCommentCount(post))
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/helpers.js]: Why this really weird "array with string" way of referencing that field on the user object?

**Comment by Discordius** [comment]
One small change, rest seems good

File: packages/lesswrong/lib/collections/posts/helpers.js (hunk 1 above)
Comment (jpaddison3) [reply packages/lesswrong/lib/collections/posts/helpers.js]: Cause I programmed in python too recently.

**Comment by jpaddison3** [comment]
@Discordius ready for re-review

**Comment by Discordius** [comment]
Will wait on merging for a bit to not coincide with the relatively risky changes we just made to the nominations system

**Comment by jpaddison3** [comment]
I bet this is now ready. Motion to merge.

File: packages/lesswrong/lib/collections/users/custom_fields.js [hunk 2]
```diff
@@ -431,6 +431,19 @@ addFieldsDict(Users, {
     order: 56,
   },
 
+  showHideKarmaOption: {
+    type: Boolean,
+    optional: false,
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/users/custom_fields.js]: Alas, this broke signup for a day :(

**Pull request merged**